### PR TITLE
TB variant filter

### PR DESCRIPTION
**Pull request by pvanheus**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by pvanheus** [comment]
This provides `tb_variant_filter`, a command line tool for filtering VCF files in many of the ways that are used in M. tuberculosis research (especially phylogenetics of M. tb).

File: recipes/spaln/patch-adddef_h [hunk 1]
```diff
@@ -1,11 +0,0 @@
---- src/adddef.h.orig	2018-08-20 21:20:46.757620508 +0200
-+++ src/adddef.h	2018-08-20 21:20:53.425569375 +0200
-@@ -39,7 +39,7 @@
- #define ERROR	(-1)
```
Comment (bgruening) [inline recipes/spaln/patch-adddef_h]: @pvanheus why have you removed this one?

Comment (pvanheus) [reply recipes/spaln/patch-adddef_h]: oops, that snuck in from other work - with the `spaln` update that patch file is no longer needed but I forgot too remove it in the associated PR, now it got into this one instead.

Comment (pvanheus) [reply recipes/spaln/patch-adddef_h]: This has been reverted with `e9b29580edbf21fd9d52be304682bb2d71c7a44c` (thanks to the guide on https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request).

**Pull request merged**